RestaurantsNearMe/Data/YelpRequest/YelpRequestProvider.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  YelpRequestProvider.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2024-01-10.
+//
+
+import Foundation
+import CoreLocation
+import Combine
+import Networking
+
+class YelpRequestProvider {
+  let activeRequest = CurrentValueSubject<YelpRequest?, Never>(nil)
+  private var cancellables = Set<AnyCancellable>()
+  
+  init(
+    locationProvider: some LocationProviding,
+    cloudKitServiceProvider: some CloudKitServiceProviding,
+    baseURLPath: StaticString
+  ) {
+    locationProvider.currentLocation
+      .map { self.translateToQueryDict(from: $0) }
+      .combineLatest(cloudKitServiceProvider.fetchedAPIKey.map { self.translateToHeader(from: $0.value)})
+      .receive(on: RunLoop.main)
+      .sink { locationParameters, authHeader in
+        let request = YelpRequest(
+          path: baseURLPath,
+          headers: authHeader,
+          parameters: locationParameters
+        )
+        self.activeRequest.send(request)
+      }.store(in: &cancellables)
```

Comment:
The current request is `invalidated` when either parts that makes up the `URL` are changed. latitude / longitude or API key